builder/jdcloud/terraform_resource.go

```diff
@@ -0,0 +1,418 @@
+package jdcloud
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+	"time"
+)
+
+/*
+	Function namely 'resource.Retry' can be quite helpful in creating resource
+	that require specific attention to its status. It is also widely used in
+	terraform-provider-jdcloud.
+
+	This time when we try to create instance for packer-jdcloud. Panic occurs
+    by importing "github.com/.../terraform/resource" :
+
+		panic: /debug/requests is already registered. You may have two independent copies of
+		golang.org/x/net/trace in your binary, trying to maintain separate state. This may
+		involve a vendored copy of golang.org/x/net/trace.
+
+	I didnt do much research on it, currently I think answer to this panic happens since
+	file : 'terraform/helper/resource/grpc_test_provider.go' referred to 'net' package, and
+	thus net.Init() was invoked both in 'terraform/resource' and 'packer'.
+
+	Currently have no idea how to solve this problem gracefully, I just copied all codes
+	related to resource.Retry to here. :P
```

Comment:
It might make sense to store these common functions in your "common.go" file above rather than in a file named after terraform.  When I first saw the name I got confused, wondering why you'd included terraform code in Packer.